internal/controller/drplacementcontrol.go

```diff
@@ -1596,6 +1596,10 @@ func updatePeers(
 
 	for pvcIdx := range vrgFromView.Status.ProtectedPVCs {
 		for policyPeerClassIdx := range policyPeerClasses {
+			if vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName == nil {
+				return peerClasses
+			}
```

Comment:
If we can return early when:

    vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName == nil

why do we need to iterate over policyPeerClassIdx? We can move the check right after line 1597.